snuba_sdk/query_visitors.py

```diff
@@ -341,10 +339,7 @@ def _combine(self, query: "main.Query", returns: Mapping[str, None]) -> None:
         # - Must have certain conditions (project, timestamp, organization etc.)
         ## SUBQUERIES
         # - outer query must only reference columns from inner query, and reference by alias
-        # - inner query must be valid
         if isinstance(query.match, main.Query):
-            self.visit(query.match)
```

Comment:
Why this change ?
Moreover, shouldn't we validate that, if the query is not a join entity alias must not be used?

Reply:
This is doing duplicate work. The combine function is called after the `_visit_match` function so the query will already be validated.

Reply:
The validation comment is valid, I'll check that.

Reply:
So instead of adding validation I decided to add a flag to the translator that will omit the entity aliases for non-join queries. The reason why I did this was to avoid the end user having to have different entity/column values (mostly entity) when they are building join vs. non-join queries. This way if the end user wants to have a static set of Entities that get reused in different queries, they only need to have one set no matter which query they are building.